template/src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { FC } from 'react';
+import { TextInputProps as TextInputBaseProps } from 'react-native';
+import styled from '@emotion/native';
+import {
+  borders,
+  color,
+  layout,
+  space,
+  flex,
+  typography,
+  textStyle,
+  BorderProps,
+  FlexProps,
+  ColorProps,
+  LayoutProps,
+  SpaceProps,
+  TextStyleProps,
+  TypographyProps,
+} from 'styled-system';
+
+import { Container } from '../Container';
+import { Text } from '../Text';
+import { colors } from '../../styles';
+
+interface TextInputProps extends TextInputBaseProps {
+  /** An optional header label to render about the input */
+  topLabel?: string;
+}
+
+type ComponentProps = TextInputProps &
+  ColorProps &
+  SpaceProps &
+  TextStyleProps &
+  TypographyProps &
+  BorderProps &
+  LayoutProps &
+  FlexProps;
+
+const Wrapper = styled(Container)`
+  ${props =>
+    props.grow &&
+    `
+      flex: 1;
+    `}
+`;
+
+const TopLabel = styled(Text)``;
+
+const Input = styled.TextInput`
+  ${flex};
+  ${borders};
+  ${color};
+  ${layout};
+  ${space};
+  ${textStyle};
+  ${typography};
+`;
+
+// NOTE: for layout and dimensioning of TextInput, wrap it in a Container
+export const TextInput: FC<ComponentProps> = ({ topLabel, multiline, ...inputProps }) => (
+  <Wrapper grow={multiline}>
+    {topLabel ? <TopLabel>{topLabel}</TopLabel> : null}
+    <Input
+      autoCapitalize="none"
+      underlineColorAndroid={colors.transparent}
+      selectionColor={colors.primary}
+      multiline={multiline}
+      {...inputProps}
+    />
+  </Wrapper>
+);
+
+// grow to the full width of its parent container
+Wrapper.defaultProps = {
+  width: '100%',
+  grow: false,
+};
+
+TopLabel.defaultProps = {
+  color: colors.gray,
+  fontSize: 2,
+  my: 0.5,
+};
+
+TextInput.defaultProps = {
```

Comment:
by leaving these here, it allows us to modify them via external props
(just for clarification of why the other 2 can be removed but this should likely stay)

Reply:
Thank you for this clarification! This is one of the questions I had about some of the button changes.